File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = new int[2];
        
        int start = 0, end = 0;
        
        int sum = sequence[0];
        
        int minLength = Integer.MAX_VALUE;
        while(end < sequence.length && start <= end){            
            while(sum < k && end < sequence.length - 1){
                sum += sequence[++end];
            }
            
            // System.out.println("Start : " + start + " " + "End : " + end + " " + "Sum : " + sum);
            
            
            if(sum == k){
                if(minLength > end - start){
                    minLength = end - start;
                    answer[0] = start;
                    answer[1] = end;
                }
            }
            sum -= sequence[start++];
            
            // System.out.println("Start : " + start + " " + "End : " + end + " " + "Sum : " + sum);
        }
        
        
        return answer;
    }
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.js
function solution(sequence, k) {
    let answer = new Array(2);
    answer[0] = 0; 
    answer[1] = 1000000;
    
    let s = 0;
    let e = 0;
    
    let sum = sequence[0];
    
    while(s <= e && e < sequence.length) {
        if(sum === k && answer[1] - answer[0] > e - s) {
            answer = [s,e];
        }
        
        if(sum > k) {
            sum -= sequence[s];
            s++;
        }else{
            e++;
            sum += sequence[e];
        }
    }
    
    return answer;
}